File: Java/Medium/WrongAnswer/find-the-divisibility-array-of-a-string_1.java
class Solution {
    public int[] divisibilityArray(String word, int m) {
        
        int n = word.length();
        int[] res = new int[n];
        int num = 0;
        for (int i = 0; i < n; i++) {
            num = (num * 10 + word.charAt(i) - '0') % m;
            res[i] = num == 0 ? 1 : 0;
        }
        return res;
    }
}

File: C++/Medium/RuntimeError/find-the-divisibility-array-of-a-string_1.cpp
class Solution {
public:
    vector<int> divisibilityArray(string word, int m) {
        
        int n = word.size();
        vector<int> ans(n);
        int r = 0;
        for (int i = 0; i < n; i++) {
            r = (r * 10 + word[i] - '0') % m;
            ans[i] = r == 0;
        }
        return ans;
    }
};